lib/controllers/dialog-controller/index.js

```diff
@@ -67,24 +67,55 @@ module.exports = class DialogController extends SkillSimulationController {
      * @param {Function} callback
      */
     setupSpecialCommands(dialogReplView, callback) {
-        dialogReplView.registerRecordCommand((filePath) => {
-            if (!stringUtils.isNonBlankString(filePath)) {
-                return callback('A file name has not been specified');
+        dialogReplView.registerRecordCommand((recordArgs) => {
+            const recordArgsList = recordArgs.trim().split(' ');
+            const RECORD_FORMAT = 'Please use the format: ".record <fileName>" or ".record <fileName> --append-quit"';
+            if (!stringUtils.isNonBlankString(recordArgs) || recordArgsList.length > 2) {
+                return Messenger.getInstance().warn(`Incorrect format. ${RECORD_FORMAT}`);
             }
-            const JSON_FILE_EXTENSION = '.json';
-            if (path.extname(filePath).toLowerCase() !== JSON_FILE_EXTENSION) {
-                return callback(`File should be of type '${JSON_FILE_EXTENSION}'`);
+            const { filePath, shouldAppendQuit } = this._validateRecordCommandInput(recordArgsList, RECORD_FORMAT);
+            const utteranceCacheCopy = [...this.utteranceCache];
+            if (shouldAppendQuit) {
+                utteranceCacheCopy.push('.quit');
             }
-            try {
-                this.createReplayFile(filePath);
-                Messenger.getInstance().info(`Created replay file at ${filePath}`);
-            } catch (err) {
-                return callback(err);
+            if (filePath) {
```

Comment:
why do you need this "if"? Should file path always be there? Should validation above fail it file path is not provided?

Reply:
in case of any failures, the helper function returns empty object. if we don't add this check. createFIle wil be called for error cases as well. 
Should filePath always be there ? Yes.